### PR TITLE
Initial peak loader

### PR DESCRIPTION
**Pull request by ant6**
- [x] find a way to read BP data and handle it (create a function?)
- [x] move from notebooks to classes
- [x] basic docstrings in BP/SOBP
- [x] test inits
- [x] implement additional helper methods (range, spread etc.)
- [x] add wercker pipelines for py2.7 and py3
- [x] cleanup and rebase
- [x] some 'default domain'

[wercker status](https://app.wercker.com/project/byKey/e724af14246bc2e21ee83eded6d0729e)
[Code Issues](https://www.quantifiedcode.com/app/project/466cefea8d324f2db542c9b47eb8cd90)

### CODE REVIEW

File: pbc/bragg_peak.py [hunk 1]
```diff
@@ -1,36 +1,116 @@
...
+        if val > val_arr.max():
+            raise ValueError('Desired values cannot be greater than max in BraggPeak!')
+        merge_idx = val_arr.argmax()
+        left = val_arr[:merge_idx]
+        right = val_arr[merge_idx:]
+        idx_left = (np.abs(left - val)).argmin()
+        idx_right = (np.abs(right - val)).argmin()
+        return idx_left, merge_idx + idx_right
+
+    def spread_at_90(self, x_arr):
```
Comment (grzanka) [inline pbc/bragg_peak.py]: It doesn't make sense to hardcode "90" into function name. Make it an optional parameter with default value 0.9 and change name of the function. Intuitively `spread` is a single return number - here you return two numbers.

File: pbc/bragg_peak.py [hunk 2]
```diff
@@ -1,36 +1,116 @@
...
+
+    def range(self, x_arr):
+        return self.spread_at_90(x_arr)[1]
+
+    def fwhm(self, x_arr):
+        """Full width af half-maximum"""
+        ll, rr = self.get_spread_idx(x_arr, 0.5)
+        return x_arr[ll], x_arr[rr]
+
+    def modulation(self, x_arr, val=0.9):
```
Comment (grzanka) [inline pbc/bragg_peak.py]: Spread and modulation concepts are not used for single Bragg peak (BP). They make sense only for SOBP (linear combination of many BPs).

**Comment by ant6** [comment]
@grzanka I think this is it for this PR, please review :)

File: pbc/bragg_peak.py [hunk 3]
```diff
@@ -1,36 +1,104 @@
...
-                            {"primary_position": self.primary_pos,
-                             "shifted_position": self.shifted_pos,
-                             "height": self.height
-                             }
-                        }
-        logger.debug("Creating BraggPeak - primary position %.2f, shifted position %.2f height %.2f\nfrom\n%s" % (
-            primary_position, shifted_position, height, bragg_peak))
+class BraggPeak(object):
+    def __init__(self, bp_domain, bp_vals):
+        if len(bp_domain) != len(bp_vals):
```
Comment (grzanka) [inline pbc/bragg_peak.py]: add some docstring explaining what users should provide as bp_domain and bp_vals.

Comment (grzanka) [inline pbc/bragg_peak.py]: are bp_domains and bp_vals saved in the BraggPeak object for future extraction >

File: pbc/bragg_peak.py [hunk 4]
```diff
@@ -1,36 +1,104 @@
...
+        else:
+            self._weight = new_weight
+
+    def evaluate(self, x_arr):
+        """Evaluate for given domain"""
+        return self._weight * self.spline(x_arr + self.initial_position - self.current_position)
 
-    def __getitem__(self, item):
-        return self.bp_data[item]
+    def get_spread_idx(self, x_arr, val):
```
Comment (grzanka) [inline pbc/bragg_peak.py]: the name of function is completely misleading. what does it have in common with spread ?

File: pbc/bragg_peak.py (hunk 3 above)
Comment (ant6) [inline pbc/bragg_peak.py]: No, they are used for `self.spline = interpolate.InterpolatedUnivariateSpline(bp_domain, bp_vals, ext=3) ` and dropped. Should I save them?

File: pbc/bragg_peak.py (hunk 4 above)
Comment (ant6) [inline pbc/bragg_peak.py]: It should be something like "get_the_longest_section_above(given_val)". Maybe `longest_section_idx(threshold)` ?

File: optimize.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    p6 = BraggPeak(x_peak, y_peak)
+    p7 = BraggPeak(x_peak, y_peak)
+    p8 = BraggPeak(x_peak, y_peak)
+    p9 = BraggPeak(x_peak, y_peak)
+    p10 = BraggPeak(x_peak, y_peak)
+    p11 = BraggPeak(x_peak, y_peak)
+    p12 = BraggPeak(x_peak, y_peak)
+    p13 = BraggPeak(x_peak, y_peak)
+    p14 = BraggPeak(x_peak, y_peak)
+    p15 = BraggPeak(x_peak, y_peak)
```
Comment (grzanka) [inline optimize.py]: C'mon - why manually ???

**Pull request merged**